source/lambda/jobresultprocessor/lambda_function.py

```diff
@@ -151,12 +151,13 @@ def processRequest(request):
     print("Processed Comprehend data: {}".format(comprehendAndMedicalEntities))
 
     # index document once the comprehend entities and KVPairs have been extracted
-    for key, val in opg_output[KVPAIRS].items():
-        if key not in comprehendAndMedicalEntities:
-            comprehendAndMedicalEntities[key] = val
-        else:
-            comprehendAndMedicalEntities[key].append(val)
-    opg.indexDocument(opg_output[DOCTEXT], comprehendAndMedicalEntities)
+    if 'ES_DOMAIN' in os.environ :
```

Comment:
Same comment as for syncprocessor

Reply:
Excellent question. Keys with undefined values are actually not considered when compiled in Typescript.
Here is an [example](https://www.typescriptlang.org/play?ssl=1&ssc=1&pln=20&pc=3#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCoASMFmgY7p7ehCTkVOle4jUMdRLYTqCc8LEZzCZmoNJODPHFZZXVtZYYkAAeRJTInDQS8po+rf40gnjbDKv8LqD2jpbYoACqAEoAMsK7sUmxkGSCc+VVQQuaTwVb1UBrDYULY7PagbgUZLJH6QbYmJAECjuMigZEMVDsJzCFLNXxtajBBCcQQ0MEAXlAAG9FCUSgAiBhs0D5UBs5IUJxs3gs1ls6RcnlcbiQfmcSB8EXs2AS3m5BRkNmKAC+lPg1Pg7gC0HgzAAFKsAJTBFQ4BoeMheRDwAYMEHsGjfNxmZjsBjMSD8WDQdSJV3wPQiLxsgCiqwYiBq+i5zt54AcqAA5MIAHKQCRsoIqADy3zIu0EAa0kGOqEcClc0jwoCcTp0oyw+hQCEQMOEV09Emd0REBg4OO6LDYUqCJSAA)

When the cfn tempalte is created, the ES_DOMAIN variable (as such as with others such as vpc) are not passed 
This is different in Javascript tho

Reply:
So I did a little bit of research on this  and even though TS/JS files do allow this behavior (and I verified that env variable is not passed), it is recommended to limit the use of undefined . Updated